### ISSUE TITLE
Input file contents should be saved in h5 output

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#1587]
We save lots of stuff, but the actual parameters of the run do not appear to be included.

**Comment by nilsdeppe** [comment]
Agreed! The current difficulty here is getting the input file to the observers. What do you think of the idea of having the option parser have a special tag `OptionTags::InputFile` that returns the input file as a string (ideally including all comments, etc.)? We can then stick that in the global cache for the observers to access. Does this sound like a reasonable plan forward?

**Comment by nilsdeppe** [comment]
Discussion with @wthrowe and @kidder:

Adding to the global cache seems reasonable and just tell people not to reparse options in random places. A little bit of smarts will need to be added to the option parser to return the string but should be fairly manageable.
